game-app/game-core/src/main/java/games/strategy/triplea/delegate/move/validation/MoveValidator.java

```diff
@@ -1171,11 +1171,10 @@ private MoveValidationResult validateTransport(
       final GamePlayer player,
       final Map<Unit, Unit> unitsToTransports,
       final MoveValidationResult result) {
-    final boolean isEditMode = getEditMode(data.getProperties());
-    if (!units.isEmpty() && units.stream().allMatch(Matches.unitIsAir())) {
+    if (!route.hasWater()) {
       return result;
     }
-    if (!route.hasWater()) {
+    if (!units.isEmpty() && units.stream().allMatch(Matches.unitIsAir())) {
```

Comment:
Is there a specific reason preventing those if cases to be combined into a single one?